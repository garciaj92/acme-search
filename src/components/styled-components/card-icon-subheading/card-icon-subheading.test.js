import React from "react";
import CardIconSubHeading from "../card-icon-subheading";

import { mount, shallow } from "enzyme";
import toJson from 'enzyme-to-json';

// Mocks
import { consoleErrorFn } from "../../../__mocks__/console.mock";

const props = {
    iconSrc: "./test-icon.svg",
    heading: "Test Heading"
};

describe("CardIconSubHeading: Test visual components", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<CardIconSubHeading {...props} />);
    });

    it("renders correctly", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("displays sets the props correctly", () => {
        expect(wrapper.find("img").first().prop("src")).toEqual(props.iconSrc);
        expect(wrapper.find("h5").first().text()).toEqual(props.heading);
    });
});

describe("CardIconSubHeading: Test prop type validation", () => {
    let consoleErrorSpy;
    beforeEach(() => {
        console.error = consoleErrorFn
        consoleErrorSpy = jest.spyOn(console, "error");
    });

    afterEach(() => {
        consoleErrorSpy.mockClear();
    });
    
    it("does not throw an error when props are provided", () => {
        shallow(<CardIconSubHeading {...props} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(0);
    });

    it("throws an error when heading prop is not provided", () => {
        const { 
            iconSrc,
        } = props;
        shallow(<CardIconSubHeading iconSrc={iconSrc} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
    });

    it("throws an error when iconSrc prop is not provided", () => {
        const { 
            heading
        } = props;
        shallow(<CardIconSubHeading heading={heading} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
    });

    it("throws an error each prop that is not a `string`", () => {
        shallow(<CardIconSubHeading iconSrc={0} heading={0} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(2);
    });
});