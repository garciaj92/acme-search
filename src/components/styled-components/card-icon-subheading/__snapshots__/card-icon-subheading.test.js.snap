// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardIconSubHeading: Test visual components renders correctly 1`] = `
<CardIconSubHeading
  heading="Test Heading"
  iconSrc="./test-icon.svg"
>
  <div
    className="d-flex justify-content-center align-items-center my-2"
  >
    <Image
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="./test-icon.svg"
      thumbnail={false}
      width={30}
    >
      <img
        className=""
        src="./test-icon.svg"
        width={30}
      />
    </Image>
    <h5
      className="mb-0 mx-1"
    >
      Test Heading
    </h5>
  </div>
</CardIconSubHeading>
`;
